src/os/bluestore/BlueStore.cc

```diff
@@ -1391,6 +1391,10 @@ void BlueStore::BufferSpace::finish_write(Cache* cache, uint64_t seq)
 {
   std::lock_guard<std::recursive_mutex> l(cache->lock);
 
+  if (writing.empty() || writing.rbegin()->seq < seq) {
```

Comment:
> E.g. please take a look at _discard() method - it might split Buffer into 2 parts and insert an additional buffer to the end of 'writing' list. Thus the ordering is broken if some entries with higher seq exist.

Then we shall fix the original logic too...

See https://github.com/ceph/ceph/pull/15911/files#diff-a9faffcf40600fd57aea5451cef5abe9L1392